### ISSUE TITLE
[Feature request] Add option to get cookies from browser

### ISSUE THREAD

**Issue by DjordjeMandic** [StefanLobbenmeier/youtube-dl-gui#38]
**Is your feature request related to a problem? Please describe.**
Cookies can only be supplied via txt file. Some browser extensions for exctracting it from browser can be malicious like [this one](https://www.reddit.com/r/youtubedl/comments/11i5vyq/psa_the_get_cookiestxt_extension_is_now_actively/) was. Option to get cookies from browser is natively supported by yt-dlp.

**Describe the solution you'd like**
In `Add authentication` window add option to get them from browser natively using yt-dlp

**Comment by StefanLobbenmeier** [comment]
Good idea! Can you link how to do this with yt-dlp

**Comment by StefanLobbenmeier** [comment]
Found it myself: --cookies-from-browser BROWSER[+KEYRING][:PROFILE][::CONTAINER]

**Comment by markjirousek** [comment]
Any news about this feature?

**Comment by StefanLobbenmeier** [comment]
Yeah it’s available in v3. Hopefully this winter I can get feature parity with v2, but feel free to already try it

**Comment by markjirousek** [comment]
I looked everywhere but didn't find where to download the v3. I guess it's supposed to be built from source but I have no idea how it works so I'll wait patiently unless I missed a link somewhere.

**Comment by StefanLobbenmeier** [comment]
You can find the precompiled versions here: https://github.com/StefanLobbenmeier/yt-dlp-compose/releases

**Comment by markjirousek** [comment]
Thank you, I’ll probably check that out one of these days when I’ll have some spare time. But, may I ask, what is yt-dlp-compose exactly? The repository of the currently in dev v3? Are you rebuilding it from scratch? Why isn’t it on the same repository?

Sorry for these questions but there are too many things I don’t understand about all this dev stuff and I’m kind of lost on GitHub, which I find difficult to understand and navigate.

I feel like I'm crawling in a maze.

**Comment by StefanLobbenmeier** [comment]
All good. Basically it’s a complete rewrite from scratch. I am not the original author of the project but just took it over so it’s hard for me to maintain it.

you can read more about it here: https://github.com/StefanLobbenmeier/youtube-dl-gui/issues/110

it’s in a separate repo until feature parity, as in that there is not much of a downgrade (and hopefully many upgrades) to v2. But since I work full time I am pretty much waiting on vacations to sink more time into it 😄

**Comment by markjirousek** [comment]
I already knew that you aren't the original developper but totally missed that post indeed. Well, that looks very interesting and promising. I find it a good thing that you are rewriting it from scratch with a more up to date architecture and technologies you are used to work with. I'm looking forward to have a look at / test / use it.

Thank you for the clarification, it all makes sense now, your patience answering my questions and above all thank you for your work.
